src/AppInstallerCLICore/Workflows/UpdateFlow.cpp

```diff
@@ -0,0 +1,135 @@
+        ManifestComparator manifestComparator(context.Args, installationMetadata);
+        bool updateFound = false;
+
+        // The version keys should have already been sorted by version
+        const auto& versionKeys = m_package.GetAvailableVersionKeys();
+        for (const auto& key : versionKeys)
+        {
+            // Check Update Version
+            if (IsUpdateVersionApplicable(context, Utility::Version(key.Version)))
+            {
+                auto manifest = m_package.GetAvailableVersion(key)->GetManifest();
+
+                // Check MinOSVersion
+                if (!manifest.MinOSVersion.empty() &&
+                    !Runtime::IsCurrentOSVersionGreaterThanOrEqual(Utility::Version(manifest.MinOSVersion)))
+                {
+                    continue;
+                }
+
+                // Check applicable Installer
+                auto installer = manifestComparator.GetPreferredInstaller(manifest);
+                if (!installer.has_value())
+                {
+                    continue;
+                }
+
+                // Since we already did installer selection, just populate the context Data
+                context.Add<Execution::Data::Manifest>(std::move(manifest));
+                context.Add<Execution::Data::Installer>(std::move(installer));
+
+                updateFound = true;
+            }
+            else
+            {
+                // Any following versions are not applicable
+                break;
+            }
+        }
+
+        if (!updateFound)
+        {
+            context.Reporter.Info() << Resource::String::UpdateNotApplicable << std::endl;
+            AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_UPDATE_NOT_APPLICABLE);
+        }
+    }
+
+    void EnsureUpdateVersionApplicable(Execution::Context& context)
+    {
+        Utility::Version updateVersion(context.Get<Execution::Data::Manifest>().Version);
+
+        if (!IsUpdateVersionApplicable(context, updateVersion))
+        {
+            context.Reporter.Info() << Resource::String::UpdateNotApplicable << std::endl;
+            AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_UPDATE_NOT_APPLICABLE);
+        }
+    }
+
+    void UpdateAllApplicable(Execution::Context& context)
+    {
+        const auto& matches = context.Get<Execution::Data::SearchResult>().Matches;
+        bool updateAllHasFailure = false;
+        for (const auto& match : matches)
+        {
+            // We want to do best effort to update all applicable updates regardless on previous update failure
+            auto updateContextPtr = context.Clone();
+            Execution::Context& updateContext = *updateContextPtr;
+            updateContext.Reporter.Info() << std::endl;
+
+            updateContext.Add<Execution::Data::InstalledPackageVersion>(match.Package->GetInstalledVersion());
+            updateContext.GetFlags() = context.GetFlags();

```

Comment:
This seems like something that should always happen, and thus be part of cloning.